skorch/tests/test_net.py

```diff
@@ -125,6 +126,51 @@ def net_pickleable(self, net_fit):
         net_fit.callbacks_ = callbacks_
         return net_clone
 
+    @pytest.mark.parametrize("copy_method", ["pickle", "copy.deepcopy"])
+    def test_train_net_after_copy(self, net_cls, module_cls, data,
+                                  copy_method):
+        # This test comes from [issue #317], and makes sure that models
+        # can be trained after copying (which is really pickling).
+        #
+        # [issue #317]:https://github.com/dnouri/skorch/issues/317
+        X, y = data
+        n1 = net_cls(module_cls)
+        n1.partial_fit(X, y, epochs=1)
+        if copy_method == "copy.deepcopy":
+            n2 = copy.deepcopy(n1)
+        elif copy_method == "pickle":
+            n2 = pickle.loads(pickle.dumps(n1))
+        else:
+            raise ValueError
+
+        # Test to make sure the parameters got copied correctly
+        close = [torch.allclose(p1, p2)
+                 for p1, p2 in zip(n1.module_.parameters(),
+                                   n2.module_.parameters())]
+        assert all(close)
+
+        # make sure the parameters change
+        # cal this twice to make sure history updates below
```

Comment:
Typo "cal"; also, why are 2 calls needed? When I comment one out, the test still passes.

Reply:
Thanks for this. I can also pass `epochs=2` to `partial_fit`, and I explained more why in the comments.